File: backend/src/main/java/com/example/backend/service/AuthService.java
package com.example.backend.service;

import java.util.List;

import com.example.backend.model.request.Account;
import com.example.backend.model.request.LoginRequest;
import com.example.backend.model.respon.AuthResponse;
import com.example.backend.model.respon.UserResponse;

public interface AuthService {
    Boolean forgotPassword(String email);
    
    Boolean resetPassword(String email, String newPassword);
    
    // Đăng nhập — trả token + info
    AuthResponse login(LoginRequest request);

    // Đăng ký — trả info user vừa tạo
    UserResponse registerUser(Account account);

    // Xoay vòng token — trả accessToken mới
    AuthResponse refreshToken(String refreshToken);

    // Đăng xuất — đưa token vào blacklist
    void logout(String accessToken);

    // Kiểm tra user có đang đăng nhập không
    boolean isLoggedIn(Long userId);

    // Thu hồi token — Admin khóa tài khoản
    void revokeToken(Long userId);

    // Tạo 10 user giả định (seed)
    List<UserResponse> seedUsers();
}

File: frontend/src/service/impl/AuthService.ts
import { http } from "../http/http";
import type { LoginRequest, AccountRequest } from "../../utils/RequestType";
import type { ApiDataResponse, AuthResponse, UserResponse } from "../../utils/ResponType";

export const login = (request: LoginRequest) => {
    return http.post<ApiDataResponse<AuthResponse>>("/auth/login", request);
};

export const register = (account: AccountRequest) => {
    return http.post<ApiDataResponse<UserResponse>>("/auth/register", account);
};

export const logout = () => {
    return http.post<ApiDataResponse<void>>("/auth/logout");
};

export const refreshToken = (token: string) => {
    return http.post<ApiDataResponse<AuthResponse>>(`/auth/refresh?refreshToken=${encodeURIComponent(token)}`);
};

export const forgotPassword = (email: string) => {
    return http.post<ApiDataResponse<boolean>>(`/auth/forgot-password?email=${encodeURIComponent(email)}`);
};

export const resetPassword = (email: string, newPassword: string) => {
    return http.post<ApiDataResponse<boolean>>(
        `/auth/reset-password?email=${encodeURIComponent(email)}&newPassword=${encodeURIComponent(newPassword)}`
    );
};

export const seedUsers = () => {
    return http.post<ApiDataResponse<UserResponse[]>>("/auth/seed");
};
